packages/build-scripts/create-github-release.ts

```diff
@@ -0,0 +1,250 @@
+                                            t.children[0].children[0].value.startsWith('Updated dependencies')
+                                        ),
+                                ),
+                        );
+                    }
+                }
+            }
+            Object.keys(tokensByVersionBumpLevel).forEach(level => {
+                if (tokensByVersionBumpLevel[level].length === 0) {
+                    delete tokensByVersionBumpLevel[level];
+                }
+            });
+            if (Object.keys(tokensByVersionBumpLevel).length === 0) {
+                return;
+            }
+            return [packageName, tokensByVersionBumpLevel] as const;
+        })
+        .filter(isDefined),
+);
+
+/**
+ * Convert from:
+ *     PACKAGE_NAME -> LEVEL -> RELEASE_NOTE_AST[]
+ * ...to:
+ *     LEVEL -> RELEASE_NOTE_TEXT -> PACKAGE_NAME[]
+ */
+const releaseNotesByLevel: Record<Level, Record<ReleaseNotesTextContent, PackageName[]>> = {};
+for (const [packageName, notesRecord] of Object.entries(releaseNotesByPackage)) {
+    Object.keys(notesRecord).forEach(level => {
+        const notes = (releaseNotesByLevel[level] ||= {});
+        notesRecord[level].forEach(note => {
+            const releaseNotesText = markdown.stringify({ children: [note], type: 'root' });
+            notes[releaseNotesText] ||= [];
+            notes[releaseNotesText].push(packageName);
+        });
+    });
+}
+
+/**
+ * Format an aggregate release note
+ */
+const releaseDate = new Date();
+const releaseDateString =
+    releaseDate.getFullYear() +
+    '-' +
+    String(releaseDate.getMonth() + 1).padStart(2, '0') +
+    '-' +
+    String(releaseDate.getDate()).padStart(2, '0');
+const aggregateReleaseNotes =
+    `# @solana/${REPO_NAME}\n\n` +
+    `## [${tag}](https://github.com/${ORG_NAME}/${REPO_NAME}/compare/${priorReleaseVersion}...${tag}) (${releaseDateString})\n\n` +
+    Object.keys(releaseNotesByLevel)
+        .toSorted((a, b) =>
+            a === b ? 0 : a === 'Major' ? -1 : b === 'Major' ? 1 : a === 'Minor' ? -1 : b === 'Minor' ? 1 : 0,
+        )
+        .flatMap(level => {
+            return [
+                `### ${level} Changes\n\n`,
+                ...Object.keys(releaseNotesByLevel[level]).map(
+                    listItem =>
+                        listItem.replace(/^\* /, '* [`' + releaseNotesByLevel[level][listItem].join('`, `') + '`] ') +
+                        '\n',
+                ),
+            ];
+        })
+        .join('');
+
+/**
+ * Create the release on GitHub
+ */
+const createReleaseParams: RestEndpointMethodTypes['repos']['createRelease']['parameters'] = {
+    body: aggregateReleaseNotes,
+    make_latest: 'true',
+    name: tag,
+    owner: ORG_NAME,
+    repo: REPO_NAME,
+    tag_name: tag,
+    // Supplying this implies that GitHub will *create* the tag specified by `tag_name`
+    target_commitish: execSync('git rev-parse HEAD').toString().trim(),
```

Comment:
This is nice, in that I don't have to have a tag and push step. It's gross in that we're shelling out because I'm too lazy to figure out how to grab the current rev from CI environment variables.